src/DiscordListener.php

```diff
@@ -174,82 +174,110 @@ public function onInteraction(InteractionReceived $event): void{
                         ], null, null, null, true)->otherwise(function(ApiRejection $rejection){
-                            $this->plugin->getLogger()->error("Failed to link account: " . $rejection->getMessage());
+                            $this->plugin->getLogger()->error("Failed to send final link success-2 response: " . $rejection->getMessage());
                         });
-                        $this->plugin->getLogger()->error("Failed to link account: " . $error);
                     });
-                }
+                    $this->plugin->getDatabase()->executeGeneric("codes.delete", ["code" => $code], null, function(SqlError $error){
+                        $this->plugin->getLogger()->error("Failed to delete used code: " . $error->getMessage());
+                    });
+                }, function(SqlError $error) use ($interaction){
+                    $this->api->interactionRespondWithMessage($interaction, null, [
+                        new Embed("❌ Failed to Link Account", "An error occurred while linking your account.", null, time(), null, new Footer("DiscordAccount v" . $this->plugin->getDescription()->getVersion())),
+                    ], null, null, null, true)->otherwise(function(ApiRejection $rejection){
+                        $this->plugin->getLogger()->error("Failed to link account: " . $rejection->getMessage());
+                    });
+                    $this->plugin->getLogger()->error("Failed to link account: " . $error);
+                });
             }
-        }, function(SqlError $error) use ($interaction){
-            $this->api->interactionRespondWithMessage($interaction, null, [
-                new Embed("❌ Failed to Link Account", "An error occurred while checking your account.", null, time(), null, new Footer("DiscordAccount v" . $this->plugin->getDescription()->getVersion())),
-            ], null, null, null, true)->otherwise(function(ApiRejection $rejection){
-                $this->plugin->getLogger()->error("Failed to send link error response: " . $rejection->getMessage());
-            });
-            $this->plugin->getLogger()->error("Failed to get/check code: " . $error);
+        }
+    }, function(SqlError $error) use ($interaction){
+        $this->api->interactionRespondWithMessage($interaction, null, [
+            new Embed("❌ Failed to Link Account", "An error occurred while checking your account.", null, time(), null, new Footer("DiscordAccount v" . $this->plugin->getDescription()->getVersion())),
+        ], null, null, null, true)->otherwise(function(ApiRejection $rejection){
+            $this->plugin->getLogger()->error("Failed to send link error response: " . $rejection->getMessage());
         });
+        $this->plugin->getLogger()->error("Failed to get/check code: " . $error);
+    });
     }
+    
+  protected function updateDiscordNickname(string $userId, string $username): void {
```

Comment:
Unrelated PR change??

Please update this PR massively, I can't review a PR whos title is 'Suiiiii' and tells me absolutely nothing about whats changed, why its changed etc.